unable to add log step

## This is a...
<!-- Check ONLY one of the following options with "x" -->
<pre><code>
[ ] Feature request
[X] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>


## The problem
<!--
Briefly describe the issue you are experiencing (or the feature you want to see implemented on Syndesis).
+ For BUGS, tell us what you were trying to do and what happened instead.
+ For NEW FEATURES, describe the _User Persona_ demanding it and its use case.
-->

Log step does not appear to work anymore, the integration runtime throws this error:

```
org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route ... because of message must be specified and not empty on: Log[]
```

## Expected behavior
<!-- Describe what the desired behavior would be, enlistin gthe acceptance criteria. -->

## Screenshot
<!--
For features/bugs tackling with UI functionality, drag and drop a screenshot depicting the desired presentation layer or supporting the UX narrative for the new functionality.
-->

```
i-webh1-11-rbxcv i-webh1 org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route -LGa5pFqdfn87sTtbaNV at: >>> pipeline -> [[SetHeader[Syndesis.STEP_ID, constant{-LGa7_9Bdfn87sTtbaNX}], Log[], process[Processor@0x40faff12]]] <<< in route: Route(-LGa5pFqdfn87sTtbaNV)[[From[servlet-1]] -> [SetHeader[... because of message must be specified and not empty on: Log[]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.util.ObjectHelper.wrapRuntimeCamelException(ObjectHelper.java:1830) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:136) ~[camel-spring-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.spring.SpringCamelContext.onApplicationEvent(SpringCamelContext.java:174) ~[camel-spring-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
i-webh1-11-rbxcv i-webh1 	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
i-webh1-11-rbxcv i-webh1 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_151]
i-webh1-11-rbxcv i-webh1 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_151]
i-webh1-11-rbxcv i-webh1 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
i-webh1-11-rbxcv i-webh1 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
i-webh1-11-rbxcv i-webh1 	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
i-webh1-11-rbxcv i-webh1 Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route -LGa5pFqdfn87sTtbaNV at: >>> pipeline -> [[SetHeader[Syndesis.STEP_ID, constant{-LGa7_9Bdfn87sTtbaNX}], Log[], process[Processor@0x40faff12]]] <<< in route: Route(-LGa5pFqdfn87sTtbaNV)[[From[servlet-1]] -> [SetHeader[... because of message must be specified and not empty on: Log[]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:1303) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:204) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.startRoute(DefaultCamelContext.java:1143) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:3729) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.doStartCamel(DefaultCamelContext.java:3443) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.access$000(DefaultCamelContext.java:209) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3251) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.doStart(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:3163) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:133) ~[camel-spring-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	... 24 common frames omitted
i-webh1-11-rbxcv i-webh1 Caused by: java.lang.IllegalArgumentException: message must be specified and not empty on: Log[]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.util.StringHelper.notEmpty(StringHelper.java:339) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.util.ObjectHelper.notEmpty(ObjectHelper.java:376) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.LogDefinition.createProcessor(LogDefinition.java:85) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.createProcessor(ProcessorDefinition.java:511) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.createOutputsProcessorImpl(ProcessorDefinition.java:474) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.createOutputsProcessor(ProcessorDefinition.java:441) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.createOutputsProcessor(ProcessorDefinition.java:185) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.createChildProcessor(ProcessorDefinition.java:204) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.PipelineDefinition.createProcessor(PipelineDefinition.java:46) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.makeProcessorImpl(ProcessorDefinition.java:562) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.makeProcessor(ProcessorDefinition.java:523) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.ProcessorDefinition.addRoutes(ProcessorDefinition.java:239) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:1300) ~[camel-core-2.21.0.fuse-000094.jar!/:2.21.0.fuse-000094]
i-webh1-11-rbxcv i-webh1 	... 36 common frames omitted

```

--- a/app/ui/src/app/common/ui-patternfly/syndesis-form-control.component.html
+++ b/app/ui/src/app/common/ui-patternfly/syndesis-form-control.component.html
@@ -29,10 +29,8 @@
 
       <label>
 
-
         <input type="checkbox"
                [attr.tabindex]="model.tabIndex"
-               [checked]="model.checked"
                [dynamicId]="bindId && model.id"
                [formControlName]="model.id"
                [name]="model.name"

--- a/app/ui/src/app/core/providers/form-factory-provider.service.ts
+++ b/app/ui/src/app/core/providers/form-factory-provider.service.ts
@@ -302,8 +302,14 @@ export class FormFactoryProviderService extends FormFactoryService {
     field: ConfiguredConfigurationProperty,
     value: any
   ) {
-    let initialValue = value || field.value || field.defaultValue;
-    initialValue = initialValue === 'false' ? false : initialValue;
+    let initialValue = false;
+    if (value !== undefined && value !== null) {
+      initialValue = !!value;
+    } else if (field.value !== undefined || field.value !== null) {
+      initialValue = !!field.value;
+    } else {
+      initialValue = !!field.defaultValue;
+    }
     return new DynamicCheckboxModel(
       {
         id: key,
@@ -312,7 +318,7 @@ export class FormFactoryProviderService extends FormFactoryService {
         controlTooltip: this.getLocalizedString(field, 'controlHint'),
         hint: this.getLocalizedString(field, 'description'),
         relation: field.relation,
-        value: !!initialValue
+        value: initialValue
       },
       {
         element: {

